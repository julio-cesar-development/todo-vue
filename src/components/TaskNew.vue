<template>
  <div class="task-new">
    <input v-model="name"
      type="text"
      class="form-input"
      placeholder="Nova tarefa"
      autofocus
      v-on:keyup.enter="addTask()"
      v-on:keyup.esc="clear()"
    >
    <button type="button" class="btn" v-on:click.stop="addTask()">+</button>
  </div>
</template>

<script>
export default {
  name: 'TaskNew',
  data() {
    return {
      name: '',
    };
  },
  mounted() {
    const input = document.querySelector('.form-input');
    if (input) {
      input.focus();
    }
  },
  methods: {
    addTask() {
      if (this.name) {
        this.$emit('task-added', { name: this.name });
        this.clear();
      }
    },
    clear() {
      this.name = '';
    },
  },
};
</script>

<style scoped>
.task-new {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  flex-wrap: wrap;
  margin-bottom: 20px;
}
</style>
