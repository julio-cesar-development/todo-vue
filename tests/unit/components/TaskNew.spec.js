import { shallowMount } from '@vue/test-utils';
import TaskNew from '@/components/TaskNew.vue';

describe('TaskNew.vue', () => {
  it('TaskNew.vue should match all specifications bellow', () => {
    const wrapper = shallowMount(TaskNew);
    expect(wrapper.isVueInstance()).toBeTruthy();

    expect(wrapper.classes('task-new')).toBe(true);

    expect(wrapper.contains('input[class="form-input"]')).toBe(true);

    expect(wrapper).toMatchSnapshot();
  });
});