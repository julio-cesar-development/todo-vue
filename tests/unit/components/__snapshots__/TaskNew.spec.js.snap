// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskNew.vue TaskNew.vue should match all specifications bellow 1`] = `<div class="task-new"><input type="text" placeholder="Nova tarefa" autofocus="autofocus" class="form-input"> <button type="button" class="btn">+</button></div>`;
